packages/wpcom-checkout/src/payment-methods/paypal.tsx

```diff
@@ -45,12 +53,23 @@ export function PaypalLabel( { labelText = null } ) {
 	);
 }
 
-export function PaypalSubmitButton( { disabled, onClick } ) {
+export function PaypalSubmitButton( {
+	disabled,
+	onClick,
+}: {
+	disabled?: boolean;
+	onClick?: ProcessPayment;
+} ): JSX.Element {
 	const { formStatus } = useFormStatus();
 	const { transactionStatus } = useTransactionStatus();
 	const [ items ] = useLineItems();
 
 	const handleButtonPress = () => {
+		if ( ! onClick ) {
```

Comment:
I'm curious why `onClick` is an optional parameter that throws if not passed.

Reply:
My guess is that you wouldn't pass `onClick` if it's disabled since it wouldn't be clicked? If it's a case like this or something similar, could it be handled through a type?

Reply:
That's a good question, and one I didn't really explain in the code. The reason is that this file's export creates a special kind of object meant to be used by the composite checkout package `CheckoutSubmitButton` which _clones_ the submit button component here and adds `disabled` and `onClick` props. If I made `onClick` required, this file would not compile since the component is rendered up above (prior to being cloned) without those props.